spec/workflow/item_batcher_spec.rb

```diff
@@ -0,0 +1,194 @@
+
+      it "returns an ItemBatcher" do
+        expect(subject).to be_kind_of(described_class)
+      end
+
+      it "sets max_items_per_batch_path" do
+        expect(subject.max_items_per_batch_path).to be_kind_of(Floe::Workflow::ReferencePath)
+        expect(subject.max_items_per_batch_path).to have_attributes(:path => ["maxBatchItems"])
+      end
+    end
+
+    context "with MaxInputBytesPerBatchPath" do
+      let(:payload) { {"MaxInputBytesPerBatchPath" => "$.batchSize"} }
+
+      it "returns an ItemBatcher" do
+        pending "implement MaxInputBytesPerBatchPath"
+
+        expect(subject).to be_kind_of(described_class)
+      end
+
+      it "sets max_input_bytes_per_batch_path" do
+        pending "implement MaxInputBytesPerBatchPath"
+
+        expect(subject.max_input_bytes_per_batch_path).to be_kind_of(Floe::Workflow::ReferencePath)
+        expect(subject.max_input_bytes_per_batch_path).to have_attributes(:path => ["batchSize"])
+      end
+    end
+
+    context "with MaxItemsPerBatch and MaxItemsPerBatchPath" do
+      let(:payload) { {"MaxItemsPerBatch" => 10, "MaxItemsPerBatchPath" => "$.maxBatchItems"} }
+
+      it "raises an exception" do
+        expect { subject }.to raise_error(Floe::InvalidWorkflowError, "Map.ItemBatcher must not specify both \"MaxItemsPerBatch\" and \"MaxItemsPerBatchPath\"")
+      end
+    end
+
+    context "with MaxInputBytesPerBatch and MaxInputBytesPerBatchPath" do
+      let(:payload) { {"MaxInputBytesPerBatch" => 1_024, "MaxInputBytesPerBatchPath" => "$.batchSize"} }
+
+      it "raises an exception" do
+        pending "implement MaxInputBytesPerBatchPath"
+
+        expect { subject }.to raise_error(Floe::InvalidWorkflowError, "Map.ItemBatcher must not specify both \"MaxInputBytesPerBatch\" and \"MaxInputBytesPerBatchPath\"")
+      end
+    end
+  end
+
+  describe "#value" do
+    let(:context) { {} }
+    let(:input)   { %w[a b c d e] }
+
+    context "with MaxItemsPerBatch" do
+      let(:payload) { {"MaxItemsPerBatch" => 2} }
+
+      it "returns in batches of 2" do
+        expect(subject.value(context, input)).to eq([{"Items" => %w[a b]}, {"Items" => %w[c d]}, {"Items" => %w[e]}])
+      end
+    end
+
+    context "with MaxItemsPerBatchPath" do
+      let(:payload)     { {"MaxItemsPerBatchPath" => "$.batchSize"} }
+      let(:state_input) { {"batchSize" => 2, "items" => input} }
+
+      it "returns in batches of 2" do
+        expect(subject.value(context, input, state_input)).to eq([{"Items" => %w[a b]}, {"Items" => %w[c d]}, {"Items" => %w[e]}])
+      end
+
+      context "with an invalid value in input" do
+        let(:state_input) { {"batchSize" => 0, "items" => input} }
+
+        it "raises an exception" do
+          expect { subject.value(context, input, state_input) }
+            .to raise_error(Floe::ExecutionError, "Map.ItemBatcher field \"MaxItemsPerBatchPath\" value \"0\" must be a positive integer")
+        end
```

Comment:
Not sure if we want a test with invalid values. (are we doing tests for these?)

e.g.:

```ruby
let(:state_input) { {"batchSize" => "a", "items" => input} }
```